File: llm_generated_code/Gemini/Python/Level1/Q33.py
n = len(prices)
    if n < 2:
        return 0
    right_max = [0] * n
    right_max[-1] = 0
    for i in range(n - 2, -1, -1):
        right_max[i] = max(right_max[i + 1], prices[i + 1] - prices[i])
    left_min = prices[0]
    max_profit = 0
    for i in range(1, n):
        max_profit = max(max_profit, prices[i] - left_min + right_max[i])
        left_min = min(left_min, prices[i])
    return max_profit

File: llm_generated_code/Gemini/CPP/Level1/Q33.cpp
#include <iostream>
#include <vector>

using namespace std;

int maxProfit(const vector<int>& prices) {
    int n = prices.size();
    if (n <= 1) {
        return 0;
    }

    vector<int> leftMax(n);
    leftMax[0] = prices[0];
    for (int i = 1; i < n; ++i) {
        leftMax[i] = min(leftMax[i - 1], prices[i]);
    }

    vector<int> rightMax(n);
    rightMax[n - 1] = prices[n - 1];
    for (int i = n - 2; i >= 0; --i) {
        rightMax[i] = max(rightMax[i + 1], prices[i]);
    }

    int maxProfit = 0;
    for (int i = 0; i < n; ++i) {
        maxProfit = max(maxProfit, rightMax[i] - leftMax[i]);
    }

    return maxProfit;
}

int main() {
    int n;
    cin >> n;

    vector<int> prices(n);
    for (int i = 0; i < n; ++i) {
        cin >> prices[i];
    }

    int maxProfit = maxProfit(prices);
    cout << maxProfit << endl;

    return 0;
}
